src/btree/bt_sync.c

```diff
@@ -109,8 +109,21 @@ __sync_ref_is_obsolete(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_ADDR *addr;
 
-    addr = ref->addr;
+    /* Ignore root pages as they can never be deleted. */
+    if (__wt_ref_is_root(ref))
+        return (false);
 
+    /* Guard against marking an in-memory page to be deleted. */
+    if (ref->state != WT_REF_DISK)
+        return (false);
+
+    /* Ignore internal pages, these are taken care of during reconciliation. */
+    if (!__wt_ref_is_leaf(session, ref))
+        return (false);
+
+    /* DON'T WE HAVE TO POTENTIALLY UNPACK THE CELL? */
+
+    addr = ref->addr;
```

Comment:
I think this looks something like:
```
addr = ref->addr;
if (__wt_off_page(parent->page, addr))
    return (false);
__wt_cell_unpack(session, parent->page, addr, vpack);
return (__wt_txn_visible_all(session, vpack->newest_stop_txn, vpack->newest_stop_ts));
```
In short, we get the `WT_REF.addr` and check to make sure it points into the page (if it doesn't, we're done, this isn't a disk reference). Then, we unpack the on-page cell and use its values.

This is a little tricky because we're not holding any kind of lock on the `WT_REF` structure, so it could be changed out from under us while we're looking at it.

Reply:
Thanks for the details. As per my understanding, I updated the code to unpack the cell and find out the details. Please review my understanding